File: python/jzoffer-22.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回从上到下每个节点值列表，例：[1,2,3]
    def PrintFromTopToBottom(self, root):
        # write code here
        if not root:
            return []
        q = [root]
        res = []
        while q:
            current = q.pop(0)
            res.append(current.val)
            if current.left != None:
                q.append(current.left)
            if current.right != None:
                q.append(current.right)
        return res

File: java/jzoffer-22.java
import java.util.ArrayList;
/**
public class TreeNode {
    int val = 0;
    TreeNode left = null;
    TreeNode right = null;

    public TreeNode(int val) {
        this.val = val;

    }

}
*/
public class Solution {
    public ArrayList<Integer> PrintFromTopToBottom(TreeNode root) {
        ArrayList<Integer> res = new ArrayList<>();
        if(root == null){
            return res;
        }
        ArrayList<TreeNode> q = new ArrayList<>();
        q.add(root);
        TreeNode current = null;
        while(q.size() != 0){
            current = q.remove(0);
            res.add(current.val);
            if(current.left != null){
                q.add(current.left);
            }
            if(current.right != null){
                q.add(current.right);
            }
        }
        return res;
    }
}